replication/logic/replication_logic.go

```diff
@@ -234,25 +234,38 @@ func NewPlanner(secsPerState *prometheus.HistogramVec, bytesReplicated *promethe
 		promBytesReplicated: bytesReplicated,
 	}
 }
-func resolveConflict(conflict error) (path []*pdu.FilesystemVersion, msg string) {
+func resolveConflict(conflict error, policy PlannerPolicy) (initialSend bool, path []*pdu.FilesystemVersion, msg string) {
 	if noCommonAncestor, ok := conflict.(*ConflictNoCommonAncestor); ok {
 		if len(noCommonAncestor.SortedReceiverVersions) == 0 {
-			// TODO this is hard-coded replication policy: most recent snapshot as source
+			// NOTE: I did not keep this in sync with listStaleFiltering, it isn't clear to me what the right thing to do is.
```

Comment:
This comment is referring to the following comment in `listStaleFiltering`

https://github.com/zrepl/zrepl/blob/1c937e58f7435c509ba97baab4cfdd9958d843ae/endpoint/endpoint_zfs_abstraction.go#L824-L828